framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +93,98 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const block = await this._sidechainAPIClient.invoke<chain.Block>('chain_getBlockByHeight', {
+			height: aggregateCommit.height,
+		});
+
+		const certificate = computeCertificateFromBlockHeader(block.header);
+		certificate.aggregationBits = block.header.aggregateCommit.aggregationBits;
+		certificate.signature = aggregateCommit.certificateSignature;
+
+		return certificate;
+	}
+
+	public async getNextCertificateFromAggregateCommits(
+		lastCertifiedHeight: number,
+		aggregateCommits: AggregateCommit[],
+	): Promise<Certificate | undefined> {
+		const block = await this._sidechainAPIClient.invoke<chain.Block>('chain_getBlockByHeight', {
```

Comment:
same here, you can get the blockheader from the blockHeaders we are storing in ChainConnectorInfo